### DISCUSSION TITLE
i18n-js v4.0.0.alpha has been released

### DISCUSSION THREAD

**Discussion by fnando** [fnando/i18n-js#609] [Announcements]
Hey everyone!

I finally put together the first alpha release of v4! Here are the main features:

- Export translations using the cli (`i18n export`).
- Automatically export translations using [listen](https://rubygems.org/gems/listen) or [guard](https://rubygems.org/gems/guard).
- JavaScript utilities implemented with TypeScript.

To install it, follow the instructions from both cli and NPM packages. More details can be found at

- CLI: https://github.com/fnando/i18n-js/tree/v4/
- NPM package: https://github.com/fnando/i18n/

Don't forget to check the sample app, available at https://github.com/fnando/i18n-js-rails-example

Please, try it out and let me know if you get errors or is under an edge case that's not covered.

**Comment by espen** [comment]
awesome @fnando. Is this a full rewrite (looks like a fresh git log)? Are there any breaking changes from v3?

**Reply by fnando** [reply]
Yes, this is a complete rewrite. I still need to write the migration guide, but the CLI does diverge quite a bit from v3. 

As for the JS part, we're good, because we were already trying to match the Rails counterpart as close as possible. There some new features (like [`I18n#store`](https://github.com/fnando/i18n#updating-translation-store), to update the store by merging in new translations), but old stuff didn't break (or so I think).

**Reply by dialex** [reply]
Have you written that migration guide? These properties `fallbacks`, `translations`, `locale` no longer exist in `i18n-js/typings/index`. We found out because our tests started failing.

**Reply by espen** [reply]
> Have you written that migration guide?

https://github.com/fnando/i18n-js/blob/main/MIGRATING_FROM_V3_TO_V4.md

**Comment by pavanjoshi914** [comment]
Js based translation exporter for i18n-js will be good in future. currently i have to use yaml loader to import translations in my webpacker based setup to dump yml files in form of js objects.

because we have to decouple or js based product in future and we were expecting rails environment independent I18n.

I think ```I18n::JS.filtered_translations.to_json ``` line uses js.rb ruby code to export translations.

except specific priorities great product :tada: !!
